gslib/commands/acl.py

```diff
@@ -340,6 +350,51 @@ class AclCommand(Command):
       },
   )
 
+  def get_gcloud_storage_args(self):
+    sub_command = self.args.pop(0)
+    if sub_command == 'get':
+      if StorageUrlFromString(self.args[0]).IsObject():
+        command_group = 'objects'
+      else:
+        command_group = 'buckets'
+      gcloud_storage_map = GcloudStorageMap(gcloud_command=[
+          'alpha', 'storage', command_group, 'describe',
+          '--format=multi(acl:format=json)'
+      ],
+                                            flag_map={})
+
+    elif sub_command == 'set':
+      # Flags must be at the start of self.args to get parsed.
+      self.ParseSubOpts()
+      if self.args[0] in CANNED_GCS_ACLS:
```

Comment:
looks like the gsutil acl command relies on checking if the argument is a file, which is much easier and also safer (prevents the failure for cases where a file name is one of the canned acl names) I think .
https://github.com/GoogleCloudPlatform/gsutil/blob/6fee45bb59c6124524d3930a5daca88a701ccd9c/gslib/command.py#L1101